src/Diagnostics.RuntimeHost/Controllers/ControllerBase.cs

```diff
@@ -1,56 +1,171 @@
-﻿using Diagnostics.ModelsAndUtils;
+﻿using Diagnostics.Logger;
+using Diagnostics.ModelsAndUtils;
+using Diagnostics.ModelsAndUtils.Attributes;
 using Diagnostics.ModelsAndUtils.Models;
+using Diagnostics.ModelsAndUtils.ScriptUtilities;
+using Diagnostics.RuntimeHost.Models;
 using Diagnostics.RuntimeHost.Services;
+using Diagnostics.RuntimeHost.Services.SourceWatcher;
 using Diagnostics.RuntimeHost.Utilities;
+using Diagnostics.Scripts;
+using Diagnostics.Scripts.Models;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Primitives;
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Reflection;
 using System.Threading.Tasks;
 
 namespace Diagnostics.RuntimeHost.Controllers
 {
     public abstract class ControllerBase : Controller
     {
-        public ControllerBase()
+        protected IStampService _stampService;
+        protected ICompilerHostClient _compilerHostClient;
+        protected ISourceWatcherService _sourceWatcherService;
+        protected IInvokerCacheService _invokerCache;
+        protected IDataSourcesConfigurationService _dataSourcesConfigService;
+
+        public ControllerBase(IStampService stampService, ICompilerHostClient compilerHostClient, ISourceWatcherService sourceWatcherService, IInvokerCacheService invokerCache, IDataSourcesConfigurationService dataSourcesConfigService)
         {
+            this._stampService = stampService;
+            this._compilerHostClient = compilerHostClient;
+            this._sourceWatcherService = sourceWatcherService;
+            this._invokerCache = invokerCache;
+            this._dataSourcesConfigService = dataSourcesConfigService;
         }
-    }
 
-    public abstract class SiteControllerBase : ControllerBase
-    {
-        protected IResourceService _resourceService;
+        protected OperationContext<TResource> PrepareContext<TResource>(TResource resource, DateTime startTime, DateTime endTime)
+            where TResource : IResource
+        {
+            this.Request.Headers.TryGetValue(HeaderConstants.RequestIdHeaderName, out StringValues requestIds);
+            this.Request.Headers.TryGetValue(HeaderConstants.InternalCallHeaderName, out StringValues internalCallHeader);
+            bool isInternalRequest = false;
+            if (internalCallHeader.Any())
+            {
+                bool.TryParse(internalCallHeader.First(), out isInternalRequest);
+            }
+
+            return new OperationContext<TResource>(
```

Comment:
A comment here about the datetime seems more relevant, now I actually see what you are doing. I don't feel super strongly but it seems to me that it makes more sense to keep it in datetime format, and then in utilities that are used in kusto queries, correctly print it for kusto